cmd/lifecycle/analyzer.go

```diff
@@ -20,62 +21,86 @@ import (
-	layersDir  string
-	skipLayers bool
-	useDaemon  bool
-
-	platform cmd.Platform
+	imageName   string
+	layersDir   string
+	orderPath   string //nolint - Platform API >= 0.7
+	runImageRef string //nolint - Platform API >= 0.7
+	stackPath   string //nolint - Platform API >= 0.7
+	useDaemon   bool
+
+	platform06     analyzeArgsPlatform06
+	additionalTags cmd.StringSlice        //nolint Platform API >= 0.7
+	docker         client.CommonAPIClient // construct if necessary before dropping privileges
+	keychain       authn.Keychain
+	platform       cmd.Platform
+}
 
-	//construct if necessary before dropping privileges
-	docker   client.CommonAPIClient
-	keychain authn.Keychain
+type analyzeArgsPlatform06 struct {
+	cacheDir      string
+	cacheImageTag string
+	groupPath     string
+	skipLayers    bool
+	cache         lifecycle.Cache
+	group         buildpack.Group
 }
 
 func (a *analyzeCmd) DefineFlags() {
 	cmd.FlagAnalyzedPath(&a.analyzedPath)
-	cmd.FlagCacheDir(&a.cacheDir)
-	cmd.FlagCacheImage(&a.cacheImageTag)
-	cmd.FlagGroupPath(&a.groupPath)
 	cmd.FlagLayersDir(&a.layersDir)
-	cmd.FlagSkipLayers(&a.skipLayers)
+	if a.platformAPIVersionGreaterThan06() {
+		cmd.FlagOrderPath(&a.orderPath)
+		cmd.FlagPreviousImage(&a.imageName)
+		cmd.FlagRunImage(&a.runImageRef)
+		cmd.FlagStackPath(&a.stackPath)
+		cmd.FlagTags(&a.additionalTags)
+	} else {
+		cmd.FlagCacheImage(&a.platform06.cacheImageTag)
+		cmd.FlagCacheDir(&a.platform06.cacheDir)
+		cmd.FlagGroupPath(&a.platform06.groupPath)
+		cmd.FlagSkipLayers(&a.platform06.skipLayers)
+	}
 	cmd.FlagUseDaemon(&a.useDaemon)
 	cmd.FlagUID(&a.uid)
 	cmd.FlagGID(&a.gid)
 }
 
 func (a *analyzeCmd) Args(nargs int, args []string) error {
-	if nargs != 1 {
-		return cmd.FailErrCode(fmt.Errorf("received %d arguments, but expected 1", nargs), cmd.CodeInvalidArgs, "parse arguments")
-	}
-	if args[0] == "" {
-		return cmd.FailErrCode(errors.New("image argument is required"), cmd.CodeInvalidArgs, "parse arguments")
+	if a.supportsImageArgument() {
+		if nargs != 1 {
+			return cmd.FailErrCode(fmt.Errorf("received %d arguments, but expected 1", nargs), cmd.CodeInvalidArgs, "parse arguments")
+		}
+		if args[0] == "" {
+			return cmd.FailErrCode(errors.New("image argument is required"), cmd.CodeInvalidArgs, "parse arguments")
+		}
+		a.imageName = args[0]
 	}
-	if a.cacheImageTag == "" && a.cacheDir == "" {
-		cmd.DefaultLogger.Warn("Not restoring cached layer metadata, no cache flag specified.")
+
+	if a.restoresLayerMetadata() {
+		if a.platform06.cacheImageTag == "" && a.platform06.cacheDir == "" {
+			cmd.DefaultLogger.Warn("Not restoring cached layer metadata, no cache flag specified.")
+		}
 	}
 
 	if a.analyzedPath == cmd.PlaceholderAnalyzedPath {
 		a.analyzedPath = cmd.DefaultAnalyzedPath(a.platform.API(), a.layersDir)
 	}
 
-	if a.groupPath == cmd.PlaceholderGroupPath {
-		a.groupPath = cmd.DefaultGroupPath(a.platform.API(), a.layersDir)
+	if a.platform06.groupPath == cmd.PlaceholderGroupPath {
```

Comment:
Should we put it under one of the the if conditions for `platform API < 0.7`?

Reply:
I thought it'll be clearer for the reader to see that this if condition is irrelevant for `platform API >= 0.7` but I don't have strong opinion on this so I'm ok with leaving it as is.